packages/@aws-cdk/aws-codedeploy/lib/base-deployment-config.ts

```diff
@@ -0,0 +1,171 @@
+   * Only applicable to ECS and Lambda deployments, and must not be specified for Server deployments.
+   * @default None
+   */
+  readonly trafficRouting?: TrafficRouting;
+
+  /**
+   * Minimum number of healthy hosts.
+   * @default None
+   */
+  readonly minimumHealthyHosts?: MinimumHealthyHosts;
+}
+
+/**
+ * The base class for ServerDeploymentConfig, EcsDeploymentConfig,
+ * and LambdaDeploymentConfig deployment configurations.
+ *
+ * @resource AWS::CodeDeploy::DeploymentConfig
+ */
+export abstract class BaseDeploymentConfig extends Resource implements IBaseDeploymentConfig {
+  /**
+   * Import a custom Deployment Configuration for a Deployment Group defined outside the CDK.
+   *
+   * @param scope the parent Construct for this new Construct
+   * @param id the logical ID of this new Construct
+   * @param deploymentConfigName the name of the referenced custom Deployment Configuration
+   * @returns a Construct representing a reference to an existing custom Deployment Configuration
+   */
+  protected static fromDeploymentConfigName(scope: Construct, id: string, deploymentConfigName: string): IBaseDeploymentConfig {
+    ignore(id);
+    const arn = Stack.of(scope).formatArn({
+      service: 'codedeploy',
+      resource: 'deploymentconfig',
+      resourceName: deploymentConfigName,
+      arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+    });
+    return {
+      deploymentConfigName: deploymentConfigName,
+      deploymentConfigArn: arn,
+    };
+  }
+
+  /**
+   * This method should be used only for static references to predefined deployment configurations,
+   * like EcsDeploymentConfig.ALL_AT_ONCE
+   * @param name the name of the referenced custom Deployment Configuration
+   * @returns a reference to an existing custom Deployment Configuration
+   */
+  protected static deploymentConfig(name: string): IBaseDeploymentConfig {
+    return {
+      deploymentConfigName: name,
+      deploymentConfigArn: arnForDeploymentConfig(name),
+    };
+  }
+
+  /**
+   * The name of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props?: BaseDeploymentConfigProps) {
+    super(scope, id, {
+      physicalName: props?.deploymentConfigName,
+    });
+
+    // Traffic routing is not applicable to Server-based deployment configs
+    if (props?.trafficRouting && (props?.computePlatform === undefined || props?.computePlatform === ComputePlatform.SERVER)) {
+      throw new Error('Traffic routing config must not be specified for a Server-base deployment configuration');
+    }
+
+    // Minimum healthy hosts is only applicable to Server-based deployment configs
+    if (props?.minimumHealthyHosts && props?.computePlatform && props?.computePlatform != ComputePlatform.SERVER) {
```

Comment:
```suggestion
    if (props?.minimumHealthyHosts && props?.computePlatform !== ComputePlatform.SERVER) {
```

Reply:
Argh I always forget strict comparison 😂

Typescript question: wouldn't your suggestion throw the error if minimumHealthyHosts is defined but computePlatform is not defined?  That's intended to be a valid case and should not throw the error

Reply:
ah ok thats true, so keep the extra `&&` in there.